src/wallet/wallet.go

```diff
@@ -1257,18 +1260,121 @@ func (w *Wallet) CreateAndSignTransactionAdvanced(params CreateTransactionParams
 	txn.SignInputs(toSign)
 	txn.UpdateHeader()
 
-	inputs := make(coin.UxArray, len(txn.In))
+	inputs := make([]UxBalance, len(txn.In))
 	for i, h := range txn.In {
-		uxOut, ok := uxaSet[h]
+		uxBalance, ok := uxbMap[h]
 		if !ok {
-			return nil, nil, errors.New("Created transaction's input is not in the UxHashSet, this should not occur")
+			return nil, nil, errors.New("Created transaction's input is not in the UxBalanceSet, this should not occur")
 		}
-		inputs[i] = uxOut
+		inputs[i] = uxBalance
+	}
+
+	if err := verifyCreatedTransactionInvariants(params, txn, inputs); err != nil {
+		logger.Criticalf("CreateAndSignTransactionAdvanced created transaction that violates invariants, aborting: %v", err)
```

Comment:
In genera we should not print error if the error will be returned, as if one day we need to call this function in CLI, the error will be print out unexpectedly.

Reply:
I would have it panic, but don't want to crash node in case this has problem.  This should never print though

Reply:
For that problem with CLI, if we implement logging configuration (#1320) we can disable